### ISSUE TITLE
get_gaps() returns an empty list on a masked array

### ISSUE THREAD

**Issue by sboltz** [obspy/obspy#2299]
Hello,

I am running ObsPy 1.1.0 in Anaconda 4.6.1 (Python 3.7.1) on Ubuntu 16.04.

I've encountered a issue where `Stream.get_gaps()` returns an empty list if there is a Trace with gaps in a masked array (see example). Should `get_gaps()` still be able to identify the gaps in these Streams?

Thanks.

```
import numpy as np                                                                                      
import obspy                                                                                            

x = np.ma.array([1, 2, 3, 4, 5, 6])                                                                     
x.mask = [0, 0, 1, 1, 0, 0]                                                                             
tr = obspy.Trace(x)                                                                                     
st = obspy.Stream([tr])                                                                                 
st.get_gaps()                                                                                           
Out[7]: []

st = st.split()                                                                                         
st.get_gaps()                                                                                           
Out[9]: 
[['',
  '',
  '',
  '',
  1970-01-01T00:00:01.000000Z,
  1970-01-01T00:00:04.000000Z,
  2.0,
  2]]
```

**Comment by d-chambers** [comment]
> Trivial to implement as well, as shown above, we just need to drop a .split() into .get_gaps() internally

Maybe a cheap check for masked arrays before calling `split` just to avoid unnecessary overhead would make sense?

@sboltz this would be very straightforward if you want to implement it.

**Comment by sboltz** [comment]
Sure, I can tackle it. It should be pretty trivial.

**Comment by megies** [comment]
Closed by #2300
